dlt/sources/rest_api/typing.py

```diff
@@ -250,6 +250,7 @@ class Endpoint(TypedDict, total=False):
     method: Optional[HTTPMethodBasic]
     params: Optional[Dict[str, Union[ResolveParamConfig, IncrementalParamConfig, Any]]]
     json: Optional[Dict[str, Any]]
+    data: Optional[Any]
```

Comment:
Here it should be something like `Optional[Union[str, Dict[str, Any], List[Tuple[str, Any]]]]` but DictValidation breaks badly. Open to fix it in a follow up pr.